src/System.Private.ServiceModel/tests/Scenarios/Client/ExpectedExceptions/ExpectedExceptionTests.cs

```diff
@@ -511,6 +511,45 @@ public static void TCP_ServiceCertFailedCustomValidate_Throw_Exception()
             ScenarioTestHelpers.CloseCommunicationObjects(factory);
         }
     }
+
+    [Fact]
+    [OuterLoop]
+    // Verify product throws MessageSecurityException when the service cert is revoked
+    public static void TCP_ServiceCertRevoked_Throw_MessageSecurityException()
+    {
+        string testString = "Hello";
+
+        NetTcpBinding binding = new NetTcpBinding();
+        binding.Security.Mode = SecurityMode.Transport;
+        binding.Security.Transport.ClientCredentialType = TcpClientCredentialType.None;
+
+        EndpointAddress endpointAddress = new EndpointAddress(new Uri(Endpoints.Tcp_RevokedServerCertResource_Address), new DnsEndpointIdentity(Endpoints.Tcp_RevokedServerCertResource_HostName));
+        ChannelFactory<IWcfService> factory = new ChannelFactory<IWcfService>(binding, endpointAddress);
+        IWcfService serviceProxy = factory.CreateChannel();
+
+        //Need to wait for 2 minutes here to ensure the CRL update are in effect as the update frequency is set to 2 minutes
+        Task.Delay(TimeSpan.FromMinutes(2)).Wait();
+        try
```

Comment:
+1 to @roncain's comment. 

I've also been tinkering with ways to deal with the CRL caching to avoid this 2 minute wait. 
However, I think that this commit should go in and these changes can go in as a later change.